src/CloudFoundry.CloudController.Test.Integration/UploadAppTest.cs

```diff
@@ -0,0 +1,322 @@
+            File.WriteAllText(Path.Combine(tempAppPath, "content-unique-clone.txt"), content1);
+            File.WriteAllText(Path.Combine(tempAppPath, "content-unique2.txt"), content2);
+            File.WriteAllText(Path.Combine(tempAppPath, "zero.txt"), "");
+            File.WriteAllText(Path.Combine(tempAppPath, "zero2.txt"), "");
+
+            CreateAppResponse app = client.Apps.CreateApp(apprequest).Result;
+
+            Guid appGuid = app.EntityMetadata.Guid;
+
+            client.Apps.Push(appGuid, tempAppPath, false).Wait();
+
+            var zipFromServerPath = Path.GetTempFileName();
+            DownloadAppZip(appGuid.ToString(), zipFromServerPath);
+            if (new FileInfo(zipFromServerPath).Length == 0)
+            {
+                Assert.Inconclusive("API endpoint doesn't support file downloads");
+            }
+
+            using (var fs = File.OpenRead(zipFromServerPath))
+            {
+                using (var arch = new ZipArchive(fs, ZipArchiveMode.Read, true))
+                {
+                    Assert.IsTrue(arch.Entries.Count() == 7);
+
+                    Assert.IsTrue(arch.Entries.First(a => a.Name == "content.txt").Length == staticContent.Length);
+                    Assert.IsTrue(arch.Entries.First(a => a.Name == "zero.txt").Length == 0);
+
+                    var contentSStream = new StreamReader(arch.Entries.First(a => a.Name == "content-Unique.txt").Open());
+                    Assert.IsTrue(contentSStream.ReadToEnd() == content1);
+                }
+            }
+
+            client.Apps.DeleteApp(appGuid).Wait();
+            Directory.Delete(tempAppPath, true);
+        }
+
+        [TestMethod]
+        public void UploadTwiceTest()
+        {
+            var staticContent = "alabala";
+            var contentPath = Path.Combine(tempAppPath, "content.zip");
+
+            using (var fs = File.OpenWrite(contentPath))
+            {
+                using (var za = new ZipArchive(fs, ZipArchiveMode.Create, true))
+                {
+                    var ze = za.CreateEntry("content.txt", CompressionLevel.NoCompression);
+                    using (var sw = new StreamWriter(ze.Open()))
+                    {
+                        sw.Write(staticContent);
+                    }
+                }
+            }
+
+
+            CreateAppResponse app = client.Apps.CreateApp(apprequest).Result;
+            Guid appGuid = app.EntityMetadata.Guid;
+
+            using (var fs = File.OpenRead(contentPath))
+            {
+                client.Apps.UploadBits(appGuid, fs, new List<FileFingerprint>() {  }).Wait();
+            }
+
+            using (var fs = File.OpenRead(contentPath))
+            {
+                client.Apps.UploadBits(appGuid, fs, new List<FileFingerprint>() { }).Wait();
+            }
+
+            var zipFromServerPath = Path.GetTempFileName();
+            DownloadAppZip(appGuid.ToString(), zipFromServerPath);
```

Comment:
Verifying the files with other methods will be very flaky and environment depended. The archive would also pass through a lot of stages and will make the asserts unreliable (staging uncompress, staging compress, dea/diego decompress, windows env is case insensitive ). Also, there is no environment agnostic way of accessing files from apps (i.e. diego doesn't have files api, not all deployments have loggregator, there is no app that work on both win or linux).  

Is there another way to signal that a test doesn't support the env? In the mean time I'll just flag them with ignore until I can find a better solution.